File: app/src/main/java/com/helin/noteapplication2/PersonDao.java
package com.helin.noteapplication2;

import android.app.Activity;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class PersonDao {
    DBHelper db;
    Context mycontext;

    public PersonDao(Context mycontext) {
        this.mycontext = mycontext;
    }

    public void register(DBHelper db, String username, String password, String name, String surname){
        SQLiteDatabase dba= db.getWritableDatabase();
        ContentValues values= new ContentValues();

        values.put(db.KEY_PERSON_USERNAME,username);
        values.put(db.KEY_PERSON_PASSWORD,password);
        values.put(db.KEY_PERSON_NAME,name);
        values.put(db.KEY_PERSON_SURNAME,surname);

        dba.insertOrThrow("person",null,values);
        dba.close();
    }
    public boolean checkUser(String username, String password) {
        db=new DBHelper(mycontext.getApplicationContext());
        String[] columns = {
                db.KEY_PERSON_USERNAME,
                db.KEY_PERSON_PASSWORD
        };
        SQLiteDatabase dba = db.getReadableDatabase();
        String selection = db.KEY_PERSON_USERNAME + " = ?" + " AND " + db.KEY_PERSON_PASSWORD + " = ?";
        String[] selectionArgs = {username, password};
        Cursor cursor = dba.query(db.TABLE_PERSON,
                columns,
                selection,
                selectionArgs,
                null,
                null,
                null);
        int cursorCount = cursor.getCount();
        cursor.close();
        db.close();
        if (cursorCount > 0) {
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/helin/noteapplicationkotlin/PersonDao.kt
package com.helin.noteapplicationkotlin

import android.content.ContentValues
import android.content.Context

class PersonDao(var mycontext: Context?) {
    var db: DBHelper? = null
    fun register(db: DBHelper?, username: String?, password: String?, name: String?, surname: String?) {
        val dba = db!!.writableDatabase
        val values = ContentValues()
        values.put(DBHelper.Companion.KEY_PERSON_USERNAME, username)
        values.put(DBHelper.Companion.KEY_PERSON_PASSWORD, password)
        values.put(DBHelper.Companion.KEY_PERSON_NAME, name)
        values.put(DBHelper.Companion.KEY_PERSON_SURNAME, surname)
        dba.insertOrThrow("person", null, values)
        dba.close()
    }

    fun checkUser(username: String, password: String): Boolean {
        db = DBHelper(mycontext!!.applicationContext)
        val columns = arrayOf<String>(
                DBHelper.Companion.KEY_PERSON_USERNAME,
                DBHelper.Companion.KEY_PERSON_PASSWORD
        )
        val dba = db!!.readableDatabase
        val selection: String = DBHelper.Companion.KEY_PERSON_USERNAME + " = ?" + " AND " + DBHelper.Companion.KEY_PERSON_PASSWORD + " = ?"
        val selectionArgs = arrayOf(username, password)
        val cursor = dba.query(DBHelper.Companion.TABLE_PERSON,
                columns,
                selection,
                selectionArgs,
                null,
                null,
                null)
        val cursorCount = cursor.count
        cursor.close()
        db!!.close()
        return if (cursorCount > 0) {
            true
        } else false
    }
}